contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--3.0.0--3.1.0.sql

```diff
@@ -1151,6 +1151,51 @@ RETURNS INTEGER AS
 'babelfishpg_tsql', 'object_id'
 LANGUAGE C STABLE;
 
+CREATE OR REPLACE PROCEDURE sys.sp_helplinkedsrvlogin(
+	IN "@rmtsrvname" sysname DEFAULT NULL,
+	IN "@locallogin" sysname DEFAULT NULL
+)
+AS $$
+BEGIN
+  IF @rmtsrvname IS NULL
+	BEGIN
+    	SELECT 
+    		s.name AS linked_server, 
+    		u.usename AS local_login, 
+    		CAST(0 as smallint) AS is_self_mapping, 
+    		l.remote_name AS remote_name 
+		FROM sys.linked_logins AS l 
+		LEFT JOIN sys.servers AS s ON l.server_id = s.server_id 
+		LEFT JOIN pg_user AS u ON l.local_principal_id = u.usesysid
+	END
+  ELSE IF @locallogin IS NULL
+	BEGIN
+    	SELECT 
+    		s.name AS linked_server, 
+    		u.usename AS local_login, 
+    		CAST(0 as smallint) AS is_self_mapping, 
+    		l.remote_name AS remote_name 
+		FROM sys.linked_logins AS l 
+		LEFT JOIN sys.servers AS s ON l.server_id = s.server_id 
+		LEFT JOIN pg_user AS u ON l.local_principal_id = u.usesysid 
+		WHERE s.name = @rmtsrvname;
+	END
+  ELSE
+	BEGIN
+		SELECT 
+    		s.name AS linked_server, 
+    		u.usename AS local_login, 
+    		CAST(0 as smallint) AS is_self_mapping, 
+    		l.remote_name AS remote_name 
+		FROM sys.linked_logins AS l 
+		LEFT JOIN sys.servers AS s ON l.server_id = s.server_id 
+		LEFT JOIN pg_user AS u ON l.local_principal_id = u.usesysid 
+		WHERE s.name = @rmtsrvname AND u.usename = @locallogin;
+	END
+END;
```

Comment:
We can club all the if..else conditions. Declare two variables server_id and local_principal_id and they both are NULL if no id exists for the passed arguments. We can then rewrite the WHERE clause like WHERE (@server_id is NULL or @server_id = s.server_id) AND (@local_principal_id is NULL or @local_principal_id = l.local_principal_id)